projects/cdk/utils/miscellaneous/obfuscate.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Obfuscates a string by replacing certain characters with a symbol, based on string length.
+ *
+ * @param value the input string to obfuscate
+ * @param symbol the symbol for obfuscation
+ * @return the obfuscated string
+ *
+ * The function determines which characters to obfuscate using a regular expression and the string's length:
+ * - 8 or more: show first 2 and last 2 characters
+ * - 4 to 7: show first and last character
+ * - Less than 4: obfuscate all characters
+ */
+export function tuiObfuscate(value: string, symbol: string): string {
+    if (!value) {
+        return value;
+    }
+
+    const match = /[\p{L}\p{N}]/gu;
+
+    let showFirst = 0;
+    let showLast = 0;
```

Comment:
seems like `showFirst` and `showLast` are always equal. Maybe combine them into one variable? `visible` for example